db/db_test.cc

```diff
@@ -5703,13 +5680,39 @@ TEST_F(DBTest, SoftLimit) {
   WriteStallListener* listener = new WriteStallListener();
   options.listeners.emplace_back(listener);
 
+  // FlushMemtable with opt.wait=true does not wait for
+  // `OnStallConditionsChanged` being called. The event listener is triggered
+  // on `JobContext::Clean`, which happens after flush result is installed.
+  // We use sync point to create a custom WaitForFlush that waits for
+  // context cleanup.
+  port::Mutex flush_mutex;
+  port::CondVar flush_cv(&flush_mutex);
+  int flush_count = 0;
+  int prev_flush_count = 0;
+  SyncPoint::GetInstance()->SetCallBack(
```

Comment:
Maybe it's a good idea to always call `ClearAllCallBacks` before setting new ones. I once saw a segfault caused by a situation as follows:
Somewhere in RocksDB we have a sync point, say, "SyncPoint1".
```
TEST_F(TestClass, Test1) {
  int local_var = 0;
  SyncPoint::GetInstance()->SetCallBack("SyncPoint1", [@local_var]() {...});
}

TEST_F(TestClass, Test2) {
  // Forgot to clear callbacks
  // Test2 hits "SyncPoint1", and tries to call the callback. The callback tries to access local_var
}
```